### PR TITLE
Lint proto files

### PR DESCRIPTION
**Pull request by DavidLawes**
## What does this change?

This PR adds a lint status check on PRs. We recently made some retrospective changes ([1](https://github.com/guardian/mobile-apps-api-models/pull/19), [2](https://github.com/guardian/mobile-apps-api-models/pull/20)) to the proto file to update the formatting. This status check will help to continually keep the proto file in accordance with Google's [style guide](https://protobuf.dev/programming-guides/style/).

I've also added instructions to the readme explaining how to install/lint the proto files manually.

## How to test

The lint check is triggered on pull request events. After raising this PR I validated:
- the lint workflow was executed
- the workflow [failed](https://github.com/guardian/mobile-apps-api-models/actions/runs/5410334798) because of lint errors
- after fixing the errors locally (`protolint -fix ./proto/.`) and pushing my changes the corresponding run was [successful](https://github.com/guardian/mobile-apps-api-models/actions/runs/5410348904)

Note that both running the workflow and executing the lint command locally returns information about any lint errors. For example, in this trivial example I updated the accent colour to camel case:

Screenshot 2023-06-29 at 10 16 27

And when running the lint command we see:

Screenshot 2023-06-29 at 10 15 50

Using the fix command (`protolint -fix ./proto/.`) changes the field name back to snake case.

### CODE REVIEW

File: proto/collection.proto [hunk 1]
```diff
@@ -1,177 +1,178 @@
...
-    optional bool boosted = 3;
-    optional bool compact = 4;
-    repeated Article sublinks = 5;
-    optional string html_fallback = 6;
-    optional Branding branding = 7;
-    optional bool premium_content = 8;
-    optional Palette sublinks_palette_light = 9;
-    optional Palette sublinks_palette_dark = 10;
+  enum CardType {
+    CARD_TYPE_ARTICLE_UNSPECIFIED = 0;
```
Comment (DavidLawes) [inline proto/collection.proto:117]: I think this is a valid change to make. If a zero value enum is defined (e.g. in the MAPI) response then I think it's automatically excluded from the data that's actually returned to the client. Naming zero value enums as UNSPECIFIED is also defined in the style guide: https://protobuf.dev/programming-guides/style/#enums

**Review by silvacb — APPROVED** [review]
Great! 👏

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Lint
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  lint_proto_files:
+    runs-on: macos-latest
```
Comment (tkgnm) [inline .github/workflows/lint.yml:9]: Not a major - but AFAIK macos is quite expensive on github (like 10x more expensive than Linux) so would this work on Linux instead?

Comment (DavidLawes) [reply .github/workflows/lint.yml:9]: Thanks for this. I'd understood that usage of github actions are [free](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions) for public repos (which this one is).

If we're talking about resources required then the macos-latest uses some [extra](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) hardware to execute, although the execution times of the corresponding action don't appear significantly slower than others in this codebase.

`macos-latest` was chosen for the convenience of being able to `brew install...` and is used in other [projects](https://github.com/search?q=org%3Aguardian%20macos-latest&type=code) too. Perhaps if we wanted to make a change we could get the tool from source by setting up the workflow with Go.

Comment (tkgnm) [reply .github/workflows/lint.yml:9]: ah okay got you - if github actions are free for public repos then that completely negates my concern so all good :)

File: proto/collection.proto [hunk 3]
```diff
@@ -1,69 +1,70 @@
...
+  string short_url = 2;
+  string uri = 3;
+  string web_uri = 4;
 }
 
 enum MediaType {
-    MEDIA_TYPE_VIDEO = 0;
-    MEDIA_TYPE_AUDIO = 1;
-    MEDIA_TYPE_IMAGE = 2;
+  MEDIA_TYPE_VIDEO_UNSPECIFIED = 0;
```
Comment (mohammad-haque) [inline proto/collection.proto:37]: @DavidLawes  just out of curiosity, why it is renamed?

File: .github/workflows/lint.yml (hunk 2 above)
Comment (mohammad-haque) [reply .github/workflows/lint.yml:9]: Ah, no worries. I see your comment below

File: proto/collection.proto (hunk 3 above)
Comment (mohammad-haque) [reply proto/collection.proto:37]: Ah, no worries. I see your comment below

Comment (DavidLawes) [reply proto/collection.proto:37]: I think this is a valid change to make. If a zero value enum is defined (e.g. in the MAPI) response then I think it's automatically excluded from the data that's actually returned to the client in the response payload. Naming zero value enums as UNSPECIFIED is also defined in the style guide: https://protobuf.dev/programming-guides/style/#enums

**Pull request merged**